staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -190,14 +190,18 @@ func (c *AvailableConditionController) sync(key string) error {
 		availableCondition.Reason = "ServiceNotFound"
 		availableCondition.Message = fmt.Sprintf("service/%s in %q is not present", apiService.Spec.Service.Name, apiService.Spec.Service.Namespace)
 		apiregistrationv1apihelper.SetAPIServiceCondition(apiService, availableCondition)
-		_, err := updateAPIServiceStatus(c.apiServiceClient, originalAPIService, apiService)
+		if _, err2 := updateAPIServiceStatus(c.apiServiceClient, originalAPIService, apiService); err2 != nil {
+			klog.Errorf("cannot update API service status: %v", err2)
+		}
 		return err
```

Comment:
this isn't correct... if the service wasn't found, and we successfully updated the status, we should return nil, not a NotFound error

Reply:
returning an error here will requeue for resync permanently, which will cause the sync method to back off on handling this apiservice